### ISSUE TITLE
Flashing script on windows not working

### ISSUE THREAD

**Issue by Oluwaseyi14** [sailfishos-sony-tama/main#170]
Flashing script on windows is stuck at searching for compatible device,  device is in fastboot mode and also using usb2 but the script is not detecting  my device.
![Script](https://user-images.githubusercontent.com/45014950/137540105-64352ce8-7925-4759-8c6a-e83e6d4364e9.png)
![Script run as administrator](https://user-images.githubusercontent.com/45014950/137540133-df7d7fd0-950a-4849-9ab7-f049521d0c9b.png)

**Comment by rinigus** [comment]
Thank you for filing it here! Maybe some Windows users can help you out.

**Comment by mrbenedetto** [comment]
I have the same Problem with my H8416. My device was running v3.4 and now I wanted to upgrade to v4.2 and I got the same issue.

Maybe Error in flash-on-windows.bat in v4.2
:: Ensure that we are flashing right device
:: :: H8416  (<<<--- :: to much, compare v3.4 .bat file)
:: H841
![Capture](https://user-images.githubusercontent.com/92653582/137604393-ac40f235-5187-4ce1-8827-5522c365c5b3.PNG)

Did not help: https://forum.sailfishos.org/t/xperia-tama-flashing-troubleshooting/7251/5

**Comment by xsetiadi** [comment]
just check windows script flash, and found this on line 169 "@call :fastboot flash oem_a %blobfilename%" does %blobfilename% refer to vendor.img001 inside sfos.zip or it is refer to vbmeta.img?

just curious, because im gona execute on windows, once my xz2 arrive from local store

**Comment by xsetiadi** [comment]
> just check windows script flash, and found this on line 169 "@call :fastboot flash oem_a %blobfilename%" does %blobfilename% refer to vendor.img001 inside sfos.zip or it is refer to vbmeta.img?
> 
> just curious, because im gona execute on windows, once my xz2 arrive from local store

ok got it,  its refer to swbinaries

**Comment by rinigus** [comment]
Did it work for you now? Assuming so from the other issue that you opened regarding MTP

**Comment by xsetiadi** [comment]
> Did it work for you now? Assuming so from the other issue that you opened regarding MTP

not yet, my xz2 will arrive tomorrow, but I will install sfos right to it. did you mean when I'm open MTP from another device it is interfering with the flashing process?

**Comment by rinigus** [comment]
Issue was opened by @Oluwaseyi14 and that person opened another issue regarding MTP in this repository. From that I conclude that flashing worked. So, question was to @Oluwaseyi14 with the original issue.

**Comment by Oluwaseyi14** [comment]
The script didn't work. I had to install and use Linux Mint.

**Comment by rinigus** [comment]
Thanks for letting me know. Let's see if someone can fix it, I am not going to work on that as I don't use Windows

**Comment by xsetiadi** [comment]
> Thanks for letting me know. Let's see if someone can fix it, I am not going to work on that as I don't use Windows

i just compare the flash-on-windows.bat from the official Xperia 10 Jolla and from your port. the difference is. on your script, there's no command to flash vendor.img001 while on Jolla script it flashes on vendor_a and vendor_b partition.

is that ok?

**Comment by Oluwaseyi14** [comment]
> > Thanks for letting me know. Let's see if someone can fix it, I am not going to work on that as I don't use Windows
> 
> i just compare the flash-on-windows.bat from the official Xperia 10 Jolla and from your port. the difference is. on your script, there's no command to flash vendor.img001 while on Jolla script it flashes on vendor_a and vendor_b partition.
> 
> is that ok?

Yes it is . we are suppose to flash vendor seperately after running the flash-on-windows.bat

**Comment by xsetiadi** [comment]
> > > Thanks for letting me know. Let's see if someone can fix it, I am not going to work on that as I don't use Windows
> > 
> > 
> > i just compare the flash-on-windows.bat from the official Xperia 10 Jolla and from your port. the difference is. on your script, there's no command to flash vendor.img001 while on Jolla script it flashes on vendor_a and vendor_b partition.
> > is that ok?
> 
> Yes it is . we are suppose to flash vendor seperately after running the flash-on-windows.bat

but you didn't mention it on flashing.md is it ok?

**Comment by rinigus** [comment]
None of the scripts (Linux or Windows) have that vendor.img001 image flashing. I'll have to read up on what is there. It is rather small and we do use LVM with /system and /vendor on our root partitions. Right now, we all use SFOS without flashing that image at all.

**Comment by xsetiadi** [comment]
> None of the scripts (Linux or Windows) have that vendor.img001 image flashing. I'll have to read up on what is there. It is rather small and we do use LVM with /system and /vendor on our root partitions. Right now, we all use SFOS without flashing that image at all.

so it's no problem I guess,

well, I think I know the problem of @Oluwaseyi14.  because they did not put fastboot.exe and other components in the sailfish x installation directory, windows are not like Linux. by default, you must put fastboot files and other files in the same directory on flash-on-windows.bat . unlike Linux, you cannot execute fastboot command from any directory (before you configure it)

**Comment by rinigus** [comment]
I don't think we need to flash `vendor.img001` image. It contains almost empty filesystem with just one empty subfolder. So, skip it and do as the script does.

**Comment by Oluwaseyi14** [comment]
> > None of the scripts (Linux or Windows) have that vendor.img001 image flashing. I'll have to read up on what is there. It is rather small and we do use LVM with /system and /vendor on our root partitions. Right now, we all use SFOS without flashing that image at all.
> 
> so it's no problem I guess,
> 
> well, I think I know the problem of @Oluwaseyi14.  because they did not put fastboot.exe and other components in the sailfish x installation directory, windows are not like Linux. by default, you must put fastboot files and other files in the same directory on flash-on-windows.bat . unlike Linux, you cannot execute fastboot command from any directory (before you configure it)

No everything is in the same folder

**Comment by xsetiadi** [comment]
> > > None of the scripts (Linux or Windows) have that vendor.img001 image flashing. I'll have to read up on what is there. It is rather small and we do use LVM with /system and /vendor on our root partitions. Right now, we all use SFOS without flashing that image at all.
> > 
> > 
> > so it's no problem I guess,
> > well, I think I know the problem of @Oluwaseyi14.  because they did not put fastboot.exe and other components in the sailfish x installation directory, windows are not like Linux. by default, you must put fastboot files and other files in the same directory on flash-on-windows.bat . unlike Linux, you cannot execute fastboot command from any directory (before you configure it)
> 
> No everything is in the same folder

then why flash-on-windows.bat don't detect the fastboot file?

while in @mrbenedetto it seems the fastboot works, but after installing sfos, the device won't detect as supposed to be written in the script. but detect as H8416, so the flash-on-windows.bat failed to verify and then cancel the flash. best to check inside the script and see what device for is that flash-on-windows.bat

**Comment by rinigus** [comment]
Make sure you download the correct image for your device. So 

- Which device do you have?
- How is it detected by fastboot?
- What is in the script in your downloaded ZIP?

**Comment by xsetiadi** [comment]
> Make sure you download the correct image for your device. So
> 
> * Which device do you have?
> * How is it detected by fastboot?
> * What is in the script in your downloaded ZIP?

if our device not listed, can we just edit the script? my xz2 is sov37 model, the same model as the guy in xda forum. they say can install it. I'm not yet got my device. and I'm super excited to try on myself

**Comment by rinigus** [comment]
Then you have to judge yourself which model is the closest to that and try to test with it. Probably flashing it manually then. But as I have no idea about these alternative models, I cannot help you much with it.

**Comment by xsetiadi** [comment]
> Then you have to judge yourself which model is the closest to that and try to test with it. Probably flashing it manually then. But as I have no idea about these alternative models, I cannot help you much with it.

ok, I have an idea about manual flash too, following the script about what file and partition need to flash

**Comment by lal883** [comment]
> > Make sure you download the correct image for your device. So
> > 
> > * Which device do you have?
> > * How is it detected by fastboot?
> > * What is in the script in your downloaded ZIP?
> 
> if our device not listed, can we just edit the script? my xz2 is sov37 model, the same model as the guy in xda forum. they say can install it. I'm not yet got my device. and I'm super excited to try on myself

I had to edit the script to get it to flash a "SO-05K" (another variant of XZ2c). With the script updated, its md5 checksum will change and the new checksum will have to be updated in the md5.lst also.

**Comment by xsetiadi** [comment]
> > > Make sure you download the correct image for your device. So
> > > 
> > > * Which device do you have?
> > > * How is it detected by fastboot?
> > > * What is in the script in your downloaded ZIP?
> > 
> > 
> > if our device not listed, can we just edit the script? my xz2 is sov37 model, the same model as the guy in xda forum. they say can install it. I'm not yet got my device. and I'm super excited to try on myself
> 
> I had to edit the script to get it to flash a "SO-05K" (another variant of XZ2c). With the script updated, its md5 checksum will change and the new checksum will have to be updated in the md5.lst also.

so I need to generate a new md5?

**Comment by rinigus** [comment]
Don't bother with the script, just flash manually. It should be fine if you can follow what to flash where.

**Comment by xsetiadi** [comment]
> Don't bother with the script, just flash manually. It should be fine if you can follow what to flash where.

thanks sir, i get my xz2 boot up sailfish now, just flash manual one per partition, and it work. script not recognize my device.

**Comment by rinigus** [comment]
script couldn't recognize the device as it is not officially AOSP supported one. great that it still works :)

**Comment by rinigus** [comment]
@xsetiadi: Do not mix the issues!

**Comment by rinigus** [comment]
As reported in #36, flashing seems to be fine in Windows. Closing here

**Comment by tordym** [comment]
I bought an XZ3 a couple of weeks ago and decided to try Sailfish on it last weekend. After carefully following the instructions provided on this site I ran into the same issue as mentioned here; the install-script stops while searching for a device. 

Fastboot command detects the device and I have the correct version of Sailfish for my phone.

As I couldn't be bothered to mess around with Windows too much I switched to Linux and there it worked without a hitch.

Is it possible that someone can provide a workaround for this issue by providing a step by step (command by command) tutorial for flashing manually on Windows?! I could extract that info from the script, but I am afraid of providing the wrong info here and accidentally helping someone to brick their phone.
